File: 1704 Determine if String Halves Are Alike.c
bool halvesAreAlike(char * s){
    int count;
    char *p1 = s;
    count = 0;
    while(s && *s != '\0') {
        if((*s == 'a') ||(*s == 'e') ||(*s == 'i') ||(*s == 'o') ||(*s == 'u') ||
            (*s == 'A') ||(*s == 'E') ||(*s == 'I') ||(*s == 'O') ||(*s == 'U')) {
            if(*p1 != 0)
                count++;
            else
                count--;
        }      
        s++;
        if(*p1 != '\0') {
            p1++;
            p1++;
        }
    }
    if(count == 0) {
        return true;
    }
    return false;
}


// Status: Accepted
// Runtime: 0 ms
// Memory Usage: 5.9 MB

File: 1704 Determine if String Halves Are Alike.go
func halvesAreAlike(s string) bool {
    l := len(s)
    x1 := 0
    x2 := 0
    for pos, x := range s{
        if pos < l/2{
            switch x{
                case 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U':
                    x1++
            }
        }else{
            switch x{
                case 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U':
                    x2++
            }
        }
    }
    return x1 == x2
}
// Status: Accepted
// Runtime: 0 ms
// Memory Usage: 2.2 MB
